Code/Network/RKManagedObjectRequestOperation.m

```diff
@@ -377,11 +377,28 @@ - (RKMappingResult *)refetchedMappingResult
         }
     } else {    
         for (NSString *destinationKeyPath in mappingInfo.relationshipMappingInfo) {
-            id relationshipValue = [object valueForKeyPath:destinationKeyPath];
+            id relationshipValue = nil;
+            // Objects in collection may have different types, so destination keypath may be not applicable to each of them
+            if([object conformsToProtocol:@protocol(NSFastEnumeration)]) {
+                NSMutableSet* results = [NSMutableSet set];
+                for (id item in object) {
+                    @try {
+                        id value = [item valueForKeyPath:destinationKeyPath];
+                        [results addObject:value];
+                    } @catch(NSException*) {
```

Comment:
If we don't catch this exception, the case described in the issue #2192 will not work. Do you think that it will be better to overwrite completely RKManagedObjectsFromObjectWithMappingInfo and RKManagedObjectsFromMappingResultWithMappingInfo?

From other side, RKGatherManagedObjectsFromObjectWithRelationshipMapping catches exceptions too